### PR TITLE
feat - landing page and minor features

### PR DESCRIPTION
**Pull request by therajatg**
1. landing page
2. msg shown if cart or wishlist is empty.
3. refactored cart and wishlist page into components

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,13 +1,12 @@
 import "./App.css";
-import { Navbar, Footer } from "./frontend/components/index";
+import { Navbar } from "./frontend/components/index";
 import { Router } from "./Router";
 
 function App() {
   return (
     <div className="App">
       <Navbar />
       <Router />
```
Comment (Mohitkumar33) [inline src/App.js:9]: I like that you put routes in a separate file

File: src/frontend/components/cartPriceCard/CartPriceCard.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { useCart } from "../../contexts/index";
+import styles from "./cartPriceCard.module.css";
+
+function CartPriceCard() {
+  const { cartItems } = useCart();
+
+  const totalOldPrice = cartItems.reduce(
```
Comment (Mohitkumar33) [inline src/frontend/components/cartPriceCard/CartPriceCard.js:7]: good use of reduce for cart calculations

File: src/frontend/helperFunctions/addToCart.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      "/api/user/cart",
+      {
+        product,
+      },
+      {
+        headers: { authorization: token },
+      }
+    );
+    if (response.status === 201) {
+      setCartItems(response.data.cart);
```
Comment (Mohitkumar33) [inline src/frontend/helperFunctions/addToCart.js:15]: This is nice that you are using a single function for setting the cart items

File: src/frontend/helperFunctions/addToCart.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    }
+  } catch (error) {
+    console.log(error);
```
Comment (Mohitkumar33) [inline src/frontend/helperFunctions/addToCart.js:18]: you can use console.error tho throw an error

File: src/frontend/helperFunctions/removeFromWishlist.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import axios from "axios";
+
+const removeFromWishlist = async (id, setWishlist, token) => {
+  try {
+    const response = await axios.delete(`/api/user/wishlist/${id}`, {
+      headers: { authorization: token },
+    });
+    setWishlist(response.data.wishlist);
+  } catch (err) {
+    console.log(err);
```
Comment (Mohitkumar33) [inline src/frontend/helperFunctions/removeFromWishlist.js:10]: console.error

File: src/frontend/helperFunctions/removeFromCart.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import axios from "axios";
+
+const removeFromCart = async (id, setCartItems, token) => {
+  try {
+    const response = await axios.delete(`/api/user/cart/${id}`, {
```
Comment (Mohitkumar33) [inline src/frontend/helperFunctions/removeFromCart.js:5]: you store all the api's in a separate file and import it here that will be easy that you dont't make spelling mistake when entering the api path

File: src/frontend/pages/home/Home.js [hunk 7]
```diff
@@ -1,11 +1,12 @@
 import "./home.css";
-import { Products, Filter } from "../../components/index";
+import { Products, Filter, Footer } from "../../components/index";
 
 function Home() {
   return (
-    <div className="sidebar-and-products">
+    <div className="filterProductAndFooter">
```
Comment (Mohitkumar33) [inline src/frontend/pages/home/Home.js:6]: footer you provided in the app.js. Then why here?

File: src/frontend/pages/login/login.js [hunk 8]
```diff
@@ -21,12 +21,11 @@ function Login() {
         type: "TOKEN",
         payload: response.data.encodedToken,
       });
-      navigate("/");
+      navigate("/home");
     } catch (err) {
       authDispatch({
```
Comment (Mohitkumar33) [inline src/frontend/pages/login/login.js:26]: please tell me regarding this auth dispath

**Review by Mohitkumar33 — APPROVED** [review]
Code is really clean. Everything is getting when I read the code. Good Job

**Review by shubhamkumar648 — APPROVED** [review]
I like your naming convention of function.reaming code look  neat and clean

File: src/frontend/pages/home/Home.js (hunk 7 above)
Comment (therajatg) [reply src/frontend/pages/home/Home.js:6]: Footer not provided in App.js

File: src/frontend/pages/login/login.js (hunk 8 above)
Comment (therajatg) [reply src/frontend/pages/login/login.js:26]: come to VC

**Pull request merged**